qbraid/runtime/aws/device.py

```diff
@@ -112,7 +113,17 @@ def transform(
                 program, str(AnalogHamiltonianSimulation), experiment_type
             )
 
-        if provider == "IONQ":
+        ## TODO: This block causes error for circuits with measurement. Needs to restructure this block
+        ## add a tighter condition on when it is triggered.
```

Comment:
This might be a blocker. Can you expand on what you mean here in terms of the errors raised for circuits with measurement?